### PR TITLE
KotOR scripts update

### PR DESCRIPTION
**Pull request by seedhartha**
This PR implements KotOR functions: createItemOnObject, actionOpenDoor, actionCloseDoor and actionMoveToObject as well as fixes a number of bugs.

### CODE REVIEW

File: src/engines/kotor/script/functions_action.cpp [hunk 1]
```diff
@@ -83,6 +83,32 @@ void Functions::actionCloseDoor(Aurora::NWScript::FunctionContext &ctx) {
...
 
+void Functions::actionMoveToObject(Aurora::NWScript::FunctionContext &ctx) {
+	Engines::KotOR::Object *caller = ObjectContainer::toObject(ctx.getCaller());
+
+	Engines::KotOR::Object *object = ObjectContainer::toObject(ctx.getParams()[0].getObject());
+	if (!object)
+		// TODO: Use PC if object tag is empty. Not sure if this is correct.
+		object = _game->getModule().getPC();
+
+	bool run = ctx.getParams()[1].getInt() != 0;
```
Comment (DrMcCoy) [inline src/engines/kotor/script/functions_action.cpp]: `unused variable ‘run’`

File: src/engines/aurora/kotorjadegui/listbox.cpp [hunk 2]
```diff
@@ -352,4 +352,14 @@ void WidgetListBox::positionItemWidgets() {
 	}
 }
 
+void WidgetListBox::applyChangesToItemWidgets() {
+	for (int i = 0; i < _itemWidgets.size(); ++i) {
```
Comment (DrMcCoy) [inline src/engines/aurora/kotorjadegui/listbox.cpp]: `comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Engines::WidgetProtoItem*>::size_type’`

(I.e. i should be a size_t)

**Comment by seedhartha** [comment]
Fixed the issues.

File: src/engines/kotor/script/event.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+private:
+	EventType _type;
+	int _userDefinedNumber;
+};
+
+} // End of namespace KotOR
+
+} // End of namespace Engines
+
+#endif ENGINES_KOTOR_SCRIPT_EVENT_H
```
Comment (DrMcCoy) [inline src/engines/kotor/script/event.h]: `extra tokens at end of #endif directive`

(That should be `#endif // ENGINES_KOTOR_SCRIPT_EVENT_H`)

**Comment by DrMcCoy** [comment]
It seems you broke putting on clothes :P 

20180624t091415

**Comment by seedhartha** [comment]
Fixed the header guard issue. As for putting the closes on, it was broken already (female characters only).

File: src/engines/kotor/script/functions_action.cpp [hunk 4]
```diff
@@ -83,6 +83,31 @@ void Functions::actionCloseDoor(Aurora::NWScript::FunctionContext &ctx) {
...
+
+	Engines::KotOR::Object *object = ObjectContainer::toObject(ctx.getParams()[0].getObject());
+	if (!object)
+		// TODO: Use PC if object tag is empty. Not sure if this is correct.
+		object = _game->getModule().getPC();
+
+	float range = ctx.getParams()[2].getFloat();
+
+	float x, y, z;
+	caller->getPosition(x, y, z);
```
Comment (DrMcCoy) [inline src/engines/kotor/script/functions_action.cpp]: You're not checking that the caller isn't 0 before dereferencing it here.

In fact, in the lower Taris cantina, tar_m03ae, when clicking on Mission, xoreos crashes here.

**Comment by seedhartha** [comment]
Should be fixed now.

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Comment by DrMcCoy** [comment]
Hey @seedhartha, just checking in. I just saw you removed your name from the "Party management" task in the KotOR roadmap. Does that mean you're stepping down from working on xoreos for now? No pressure or anything, only seeing if there's something I can do.

In either case, thank you for your work on xoreos so far! It's exciting to see KotOR progress. :)

**Comment by seedhartha** [comment]
Hi! Yes, I decided to take a little break from the project. Not as in stepping down though. I am still looking forward to getting KotOR games to a somewhat playable state :)

**Comment by DrMcCoy** [comment]
Okay, enjoy your break :)

**Pull request merged**